ios/Classes/SwiftMparticleFlutterSdkPlugin.swift

```diff
@@ -244,6 +244,121 @@ public class SwiftMparticleFlutterSdkPlugin: NSObject, FlutterPlugin {
+                    attributes.affiliation = affiliation
+                }
+                if let couponCode = rawTransactionAttributes["couponCode"] as? String {
+                    attributes.couponCode = couponCode
+                }
+                if let shipping = rawTransactionAttributes["shipping"] as? NSNumber {
+                    attributes.shipping = shipping
+                }
+                if let tax = rawTransactionAttributes["tax"] as? NSNumber {
+                    attributes.tax = tax
+                }
+                if let revenue = rawTransactionAttributes["revenue"] as? NSNumber {
+                    attributes.revenue = revenue
+                }
+                if let transactionId = rawTransactionAttributes["transactionId"] as? String {
+                    attributes.transactionId = transactionId
+                }
+
+                event.transactionAttributes = attributes
+            }
+
+            if let rawImpressions = callArguments["impressions"] as? [[String: Any]] {
+                for rawImpression in rawImpressions {
+                    if let listName = rawImpression["impressionListName"] as? String,
+                       let rawProducts = rawImpression["products"] as? [[String: Any]] {
+                        for rawProduct in rawProducts {
+                            if let name = rawProduct["name"] as? String,
+                               let sku = rawProduct["sku"] as? String,
+                               let quantity = rawProduct["quantity"] as? NSNumber,
+                               let price = rawProduct["price"] as? NSNumber {
+                                let newProduct = MPProduct.init(name: name,
+                                                                sku: sku,
+                                                                quantity: quantity,
+                                                                price: price)
+                                event.addImpression(newProduct, listName: listName)
+                            }
+                        }
+                    }
+                }
+            }
+
+            // Optionable Properties on MPCommerceEvent
+            if let checkoutOptions = callArguments["checkoutOptions"] as? String {
+                event.checkoutOptions = checkoutOptions
+            }
+            if let currency = callArguments["currency"] as? String {
+                event.currency = currency
+            }
+            if let productListName = callArguments["productListName"] as? String {
+                event.productListName = productListName
+            }
+            if let productListSource = callArguments["productListSource"] as? String {
+                event.productListSource = productListSource
+            }
+            if let screenName = callArguments["screenName"] as? String {
+                event.screenName = screenName
+            }
+            // ??? checkoutStep is deprecated for iOS
```

Comment:
huh, it's not deprecated on Android, but I imagine it probably should be... @rmi22186, do you think it's worth getting some clarification on whether we should include `checkoutStep` on Flutter at all or just go with it for now?

Reply:
It's not deprecated on web either.  I'll escalate this one to see what the history is here.

Reply:
I'm leaning that we should include it but I just wanted to call it out just in case.